File: JobHunter.Domain/Models/VacancyFiltersModels/FilterModel.cs
using System.Collections.Generic;

namespace JobHunter.Domain.Models.VacancyFiltersModels
{
    public class FilterModel
    {
        public string Query { get; set; }
        public string City { get; set; }
        public Gender Gender { get; set; }
        public Language Language { get; set; }
        public string Experience { get; set; }
        public Accommodation Accommodation { get; set; }
        public decimal VacancyCost { get; set; }
        public Visa Visa { get; set; }
        public int CurrentPage { get; set; }
        public List <CategorySelected> Categories{ get; set; }

    }

    public class CategorySelected
    {
        public string CategoryName { get; set; }
        public bool IsSelected { get; set; }
    }
}

File: JobHunter.Presentation/ClientApp/src/app/models/FilterModel.ts
import { FilterModel, FilterEndPoints } from '../interfaces/vacancyListInterfaces';


export const filterModel: FilterModel =  {
    query: '',
  city: '',
  languageLevels: [],
  categories: [],
    gender: {
      male: false,
      female: false,
      other: false
    },
    language: {
      without: false,
      beginer: false,
      elementary: false,
      intermediate: false,
      advanced: false
    },
     experience: '',
    accomodation: {
      without: false,
      free: false,
      paid: false
    },
    vacancyCost: 0,
    visa: {
      on: false,
      off: false
  },
  currentPage: 1,
};

export const filterEndPoints: FilterEndPoints = {
  languageLevels: [],
  categories: [],
  experience: [],
  salaryMinValue: 0,
  salaryMaxValue: 0
};
